mtags/src/main/scala-3/scala/meta/internal/pc/completions/MatchCaseCompletions.scala

```diff
@@ -86,34 +86,36 @@ object CaseKeywordCompletion:
             new Parents(argTypes, definitions)
           case _ =>
             new Parents(NoType, definitions)
-      case sel => new Parents(sel.tpe, definitions)
+      case sel =>
+        val selectorTpe = sel.tpe
+        new Parents(selectorTpe, definitions)
 
     val selectorSym = parents.selector.typeSymbol
-
     // Special handle case when selector is a tuple or `FunctionN`.
     if definitions.isTupleClass(selectorSym) || definitions.isFunctionClass(
         selectorSym
       )
     then
-      val label =
-        if patternOnly.isEmpty then s"case ${parents.selector.show} =>"
-        else parents.selector.show
-      List(
-        CompletionValue.CaseKeyword(
-          selectorSym,
-          label,
-          Some(
-            if patternOnly.isEmpty then
+      if patternOnly.isEmpty then
+        val selectorTpe = parents.selector.show
+        val tpeLabel =
+          if !selectorTpe.contains("x$1") then selectorTpe
```

Comment:
`x$1` is actually a correct name :thinking: We should probably make sure that we are in an anonymous function and not a pattern match?

Reply:
I thought that using names with `$` is undefined behaviour, so it's better for the users not to use such names?

Reply:
> We should probably make sure that we are in an anonymous function and not a pattern match?
 I'm not sure what you mean?